script/eslint.ts

```diff
@@ -1,35 +0,0 @@
-#!/usr/bin/env ts-node
-
-import * as Path from 'path'
-import chalk from 'chalk'
-
-const { CLIEngine } = require('eslint')
-
-const shouldFix = process.argv.indexOf('--fix') > -1
-const eslint = new CLIEngine({
-  cache: true,
-  cwd: Path.dirname(__dirname),
-  fix: shouldFix,
-  rulePaths: [Path.join(__dirname, '..', 'eslint-rules')],
-})
-
-const report = eslint.executeOnFiles([
-  './{script,eslint-rules}/**/*.{j,t}s?(x)',
-  './tslint-rules/**/*.ts',
-  './app/*.ts',
-  './app/{src,typings,test}/**/*.{j,t}s?(x)',
-  './changelog.json',
-])
-
-if (shouldFix) {
-  CLIEngine.outputFixes(report)
-}
-
-console.log(eslint.getFormatter()(report.results))
-
-if (report.errorCount > 0) {
-  process.exitCode = 1
-  console.error(
-    chalk`{bold.green → To fix some of these errors, run {underline yarn eslint --fix}}`
```

Comment:
This functionality seems to be removed. Is this line useful or is it OK if it goes away?

Reply:
If we're going to take this hint away I'd love a general purpose linting doc (however brief) that shows the options available and how to use them, that we can point contributors to. The [Working with Packages](https://github.com/desktop/desktop/blob/master/docs/contributing/working-with-packages.md)  doc is a good example of niche tooling information.

Thoughts @outofambit ?

Reply:
eslint's autofix stuff has been a bit hit or miss for me so I generally don't use it, and won't really miss the script going away, but this is pretty self-explanatory about what to do next:

```shellsession
$ yarn lint:src
yarn run v1.10.1
$ yarn tslint && yarn eslint-check && yarn eslint
$ tslint -p .
$ eslint --print-config .eslintrc.* | eslint-config-prettier-check
No rules that are unnecessary or conflict with Prettier were found.
$ eslint --cache --rulesdir ./eslint-rules "./{script,eslint-rules,tslint-rules}/**/*.{j,t}s{,x}" "./app/{src,typings,test}/**/*.{j,t}s{,x}" "./changelog.json"

/Users/shiftkey/src/desktop/app/src/lib/git/pull.ts
  78:27  error  Expected '===' and instead saw '=='  eqeqeq

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
>  I'd love a general purpose linting doc

i think that's a great idea. i'll add one to this pr

Reply:
added in 540248a